torchaudio/prototype/models/hifi_gan.py

```diff
@@ -0,0 +1,307 @@
+    """
+
+    def __init__(
+        self, channels: int, kernel_size: int = 3, dilation: Tuple[int, int] = (1, 3), lrelu_slope: float = 0.1
+    ):
+        super(ResBlock2, self).__init__()
+        self.convs = nn.ModuleList(
+            [
+                Conv1d(
+                    channels,
+                    channels,
+                    kernel_size,
+                    1,
+                    dilation=dilation[0],
+                    padding=get_padding(kernel_size, dilation[0]),
+                ),
+                Conv1d(
+                    channels,
+                    channels,
+                    kernel_size,
+                    1,
+                    dilation=dilation[1],
+                    padding=get_padding(kernel_size, dilation[1]),
+                ),
+            ]
+        )
+        self.lrelu_slope = lrelu_slope
+
+    def forward(self, x: torch.Tensor):
+        """
+        Args:
+            x (Tensor): input of shape ``(batch_size, channels, time_length)``.
+        Returns:
+            Tensor of the same shape as input.
+        """
+        for c in self.convs:
+            xt = F.leaky_relu(x, self.lrelu_slope)
+            xt = c(xt)
+            x = xt + x
+        return x
+
+
+def get_padding(kernel_size, dilation=1):
+    """Find padding for which 1D convolution preserves the input shape."""
+    return int((kernel_size * dilation - dilation) / 2)
+
+
+def hifigan_model(
+    in_channels: int,
+    upsample_rates: Tuple[int, ...],
+    upsample_initial_channel: int,
+    upsample_kernel_sizes: Tuple[int, ...],
+    resblock_kernel_sizes: Tuple[int, ...],
+    resblock_dilation_sizes: Tuple[Tuple[int, ...], ...],
+    resblock_type: int,
+    lrelu_slope: float,
+) -> HiFiGANGenerator:
+    r"""Builds HiFi GAN Generator :cite:`NEURIPS2020_c5d73680`.
+    Args:
+        in_channels (int): See :py:class:`HiFiGANGenerator`.
+        upsample_rates: See :py:class:`HiFiGANGenerator`.
+        upsample_rates (tuple of int): See :py:class:`HiFiGANGenerator`.
+        upsample_initial_channel (int): See :py:class:`HiFiGANGenerator`.
+        upsample_kernel_sizes (tuple of int): See :py:class:`HiFiGANGenerator`.
+        resblock_kernel_sizes (tuple of int): See :py:class:`HiFiGANGenerator`.
+        resblock_dilation_sizes (tuple of tuples of int): See :py:class:`HiFiGANGenerator`.
+        resblock_type (int, 1 or 2): See :py:class:`HiFiGANGenerator`.
+    Returns:
+        HiFiGANGenerator: generated model.
+    """
+
+    return HiFiGANGenerator(
+        upsample_rates=upsample_rates,
+        resblock_kernel_sizes=resblock_kernel_sizes,
+        resblock_dilation_sizes=resblock_dilation_sizes,
+        resblock_type=resblock_type,
+        upsample_initial_channel=upsample_initial_channel,
+        upsample_kernel_sizes=upsample_kernel_sizes,
+        in_channels=in_channels,
+        lrelu_slope=lrelu_slope,
+    )
+
+
+def hifigan_v1() -> HiFiGANGenerator:
```

Comment:
same here, maybe call it `hifigan_generator_v1`?

Reply:
Done